### ISSUE TITLE
How to reproduce results in paper

### ISSUE THREAD

**Issue by Luciennnnnnn** [haoyuc/MaskedDenoising#7]
Hi, authors, great work!

Does the results in paper is produced by this [configuration](https://github.com/haoyuc/MaskedDenoising/blob/main/options/masked_denoising/input_mask_80_90.json)? Should I train it with 4 GPUs to achieve same results? 1 GPU or 4 GPUs, which one is official setting?

Currently, I train models with 1 GPU and 200K as the paper stated, however, I found disable input masking achieve better results in validation set you provided. Specifically, for SwinIR, the validation PSNR degrade slowly, for SwinIR with masking training, the validation PSNR grows slowly. See the figure below:

<img width="635" alt="image" src="https://github.com/haoyuc/MaskedDenoising/assets/20135317/aee09c67-487f-4343-9765-aeb132cd4dcb">

**Comment by haoyuc** [comment]
Hi, what batch size and training dataset are you using? We use a batch size of 64 and the training dataset consists of DIV2K, Flickr2K, BSD500, and WED, with over 8000 training images. 
Different batch sizes and training data yield different results. Larger batch sizes and more training images may lead to better results. 
Additionally, you may try training for a longer duration, as observed from the image you provided, the PSNR curve is still ascending.

**Comment by haoyuc** [comment]
Hi, what batch size and training dataset are you using? 
We use a batch size of **64** and the training dataset consists of **DIV2K, Flickr2K, BSD500, and WED, with over 8000 training images.**
**Different batch sizes and training data yield different results. Larger batch sizes and more training images may lead to better results.** Additionally, you may try training for a longer duration, as observed from the image you provided, the PSNR curve is still ascending.

**Comment by Luciennnnnnn** [comment]
Hi, I use batch size 64 as specific by your configuration file. What's more, I use same dataset as you, that is DIV2K, Flickr2K, BSD500, and WED. The only exception is I crop DIV2K and Flickr2K to small patches to speed up data loading process, which is a common practice used by Real-ESRGAN.

Now, I'm training models with 4 GPUs, a total of batch size 256, the results are shown below:
<img width="331" alt="image" src="https://github.com/haoyuc/MaskedDenoising/assets/20135317/cdbc732f-38c0-4587-98e1-4ffd89c9579a">
I do not see much different than single GPU training, I may train models with 300K iterations to see how them perform.

**Comment by Luciennnnnnn** [comment]
Hi, I have tried to train 300K iterations and validation on SIDD also.

See the figure below, light blue curve represents naive SwinIR:
<img width="624" alt="image" src="https://github.com/haoyuc/MaskedDenoising/assets/20135317/000d14fc-3dd4-4d7f-b575-33c30f0f4f2b">

For both validation set, the best performing model is observed in early iteration of naive SwinIR model, the one trained with masking strategy can not surpass it finally (increasing trend is disappear in the end of training).

Considering the performance of two models in the end of training, naive one performed better than SwinIR trained with masking strategy in McM, and performed bad than another in SIDD.

@haoyuc  Could you please help me train a model to see the benefits of masking strategy?

**Comment by haoyuc** [comment]
Firstly, we appreciate your in-depth engagement with our work.

1) Regarding _"the best performing model is observed in early iteration of naive SwinIR model"_:

It's important to clarify that our evaluation of model performance is based on the results from the final, stable stage of training, not the peak performance observed in early training iterations. This approach is predicated on the practical application context: in real-world scenarios, we often lack detailed knowledge about the degradation distribution of test images, and performance testing is often unfeasible. Therefore, we typically use models that have reached a state of convergence and avoid deploying models that have only undergone very limited training.

2) Experiment on McM:

As your experiment rightly shows, the masked training significantly outperforms the baseline model upon convergence for McM (Poison, 2). This advantage holds true across other noise categories and levels. **Our method excels in situations where there is a significant disparity between the noise distribution of the training set and that of the test set.** Therefore, when the noise level is intensified, the advantage of our method becomes even more evident, as demonstrated in our paper.

3) Experiment on SIDD:

Regarding the SIDD experiments in our supplementary materials: _"In order to simulate a scenario with extremely limited training samples, the training set only contains two 4K noisy – clean image pairs from SIDD"_.  
In the experiments you conducted, it seems that you utilized all the training images from SIDD, leading to the observed results. 
It's important to emphasize that our method is highly applicable in certain extreme real-world situations, such as 
- when the number of available training set image pairs is minimal or 
- when there is a significant disparity between the degradation distributions of the training and test sets.

We hope these responses address your queries. If you have any additional questions, let‘s have further discussions.

**Comment by Maoeyu** [comment]
![image](https://github.com/haoyuc/MaskedDenoising/assets/93516514/b6885f64-a7e6-4089-a8d9-ca1fbf675f7a)
Hello, I encountered the above error while training this model. Could it be possible that the dataroot_H path is incorrect?

**Comment by haoyuc** [comment]
@Maoeyu Hi, you need to change "dataroot_H" to your own data path.
